source/loop.cpp

```diff
@@ -420,12 +546,60 @@ void Loop::CalculateAbundanceCorrection(double helium_to_hydrogen_ratio)
   parameters.ion_mass_correction = (1.0 + 4.0*helium_to_hydrogen_ratio)/(2.0 + 3.0*helium_to_hydrogen_ratio)*(1.0 + z_avg)/z_avg;
 }
 
+void Loop::ReadRadiativeLossData()
+{
+    // Reads in the radiative loss files.  Only need to do once during the setup.
+   std::string path = "data/radiation/";
+   std::string filenames[] = {"abund_10_rad_loss.dat","abund_15_rad_loss.dat","abund_20_rad_loss.dat","abund_25_rad_loss.dat",
+                                "abund_30_rad_loss.dat","abund_35_rad_loss.dat","abund_40_rad_loss.dat"};
+   std::ifstream fin;
+   int n_files = sizeof(filenames)/sizeof(filenames[0]);
+   
+   double number;
+   char comma;
+   
+   for (int i=0; i < n_files; ++i)  // Loop over files for different abundances
+   {
+       fin.open(path+filenames[i]);
+       
+       for (int j=0; j < 100; ++j)  // Loop over temperatures (rows in files)
+       {
+           
+           for (int k=0; k < 8; ++k)   // Loop over densities (columns in files)
```

Comment:
As before, hardcoding these lengths makes me really nervous.

Reply:
Let me see if there's a way to pull the array sizes directly from the files.  This is certainly easier in Python...

Reply:
I've removed all the hard-coded values.  I need to validate the output, but that's enough for now.  It's at least running without NaNs, so it has that going for it, which is nice.

Reply:
All the plots in the paper were reproduced with the update.